src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/PreserveTlsProvider.cs

```diff
@@ -0,0 +1,137 @@
+﻿using System;
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+using Mono.Tuner;
+
+namespace MonoDroid.Tuner
+{
+	public class PreserveTlsProvider : BaseSubStep
+	{
+		public string TlsProviderType { get; set; }
+
+		public override bool IsActiveFor (Mono.Cecil.AssemblyDefinition assembly)
+		{
+			return TlsProviderType != null && (assembly.Name.Name == "System" || assembly.Name.Name == "Mono.Android");
+		}
+
+		public override SubStepTargets Targets {
+			get { return SubStepTargets.Method; }
+		}
+
+		public override void ProcessMethod (MethodDefinition method)
+		{
+			if (method.Name == "CreateDefaultProviderImpl" && method.DeclaringType.FullName == "Mono.Net.Security.MonoTlsProviderFactory")
+				ProcessCreateProviderImpl (method);
+		}
+
```

Comment:
Following 5 methods are duplicated from PreserveHttpAndroidClientHandler, I think we might need a helper class here to avoid duplicating these.